notebooks/py/MERFISH.py

```diff
@@ -0,0 +1,252 @@
+#
+#Please see [documentation](https://spacetx-starfish.readthedocs.io/en/stable/index.html) for detailed descriptions of all the data structures and methods used here.
+# EPY: END markdown
+
+# EPY: START code
+# EPY: ESCAPE %load_ext autoreload
+# EPY: ESCAPE %autoreload 2
+# EPY: END code
+
+# EPY: START code
+# EPY: ESCAPE %matplotlib inline
+
+import pprint
+import os
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from showit import image as show_image
+
+from starfish import display
+from starfish import data, FieldOfView
+from starfish.types import Features, Axes
+
+from starfish.util.plot import (
+    imshow_plane, intensity_histogram, overlay_spot_calls
+)
+# EPY: END code
+
+# EPY: START markdown
+### Load Data into Starfish from the Cloud
+#
+#The data from one field of view correspond to 16 images from 8 hybridzation rounds (r)  2 color channels (c) one z plane (z). Each image is 2048X2048 (y,x). These data correspond to U2OS cells in culture. O(100) transcripts are barcoded for subsequent spatial resolution. Pixel intensity values for one pixel, across all rounds and channels, can be decoded into the nearest barcode, thus resolving each pixel into a particular gene.
+# EPY: END markdown
+
+# EPY: START code
+use_test_data = os.getenv("USE_TEST_DATA") is not None
+
+# An experiment contains a codebook, primary images, and auxiliary images
+experiment = data.MERFISH(use_test_data=use_test_data)
+
+pp = pprint.PrettyPrinter(indent=2)
+pp.pprint(experiment._src_doc)
+
+# note the structure of the 5D tensor containing the raw imaging data
+imgs = experiment.fov().get_image(FieldOfView.PRIMARY_IMAGES)
+print(imgs)
+# EPY: END code
+
+# EPY: START markdown
+### Visualize codebook
+#
+#The MERFISH codebook maps each barcode to a gene (or blank) feature. The barcodes are 16 bit vectors that can be read out, for each pixel, from the 8 rounds and 2 color channels. The codebook contains a precise specificaton of how each of these 16 bit barcode vectors relate to the 5D tensor of raw image data
+# EPY: END markdown
+
+# EPY: START code
+experiment.codebook
+# EPY: END code
+
+# EPY: START markdown
+### Visualize raw data
+#
+#A nice way to page through all this data is to use the ```display``` command. We have commented this out for now, because it will not render in Github. Instead, we simply show an image from the first round and color channel.
+# EPY: END markdown
+
+# EPY: START code
+# Display all the data in an interactive pop-up window. Uncomment to have this version work.
+
+# %gui qt5
+# display(imgs)
+
+# Display a single plane of data
```

Comment:
ah, first and foremost, i didn't quite understand how **kwargs were propagated from imshow_plane to the underlying call to im_show. second, not here but in other examples, I needed to pass a data driven call to clim, e.g., ```clim = list(np.percentile(single_plane.data, [5, 99])``` which happens after the imagestack is sliced into a numpy array. adding this functionality into ```imshow_plane``` would be awesome! the api should be more like percentiles=None, but if you specify these values it will compute the percentiles appropriately and wrap them up in the clim kwarg.